pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -369,6 +369,7 @@ func (t *Tracer) listen(
 			eventCallback(types.Base(eventtypes.Err(err.Error())))
 			continue
 		}
+		event.Key = key
```

Comment:
Another advantage of using the `key` like this (namespace/pod or containerID) is that we are still able to enrich the events for this kind of gadget in `local-gadget` without having to use [this](https://github.com/inspektor-gadget/inspektor-gadget/blob/7ba006c31b5cf51ec0ded76570ab17bcaf143813/pkg/container-collection/options.go#L54-L61) fake information (which I was working to remove). I will check this approach for other gadgets.